### PR TITLE
MRG : Evoked grand average 3

### PR DESCRIPTION
**Pull request by agramfort**
cc @dengemann see fix of interpolation code

### CODE REVIEW

**Comment by agramfort** [comment]
not sure where to add the new function in python_reference.rst file

any suggestion?

File: mne/evoked.py [hunk 1]
```diff
@@ -938,6 +940,63 @@ def _get_evoked_node(fname):
     return evoked_node
 
 
+def grand_average(all_evokeds, interpolate_bads='eeg'):
+    """Make grand average of a list evoked data
+
+    The function interpolates bad channels based on `interpolate_bads`
+    parameter. If `interpolate_bads` is equal to 'eeg' the grand average
+    file will only contain the MEG channels that are marked good
+    in all of the evoked datasets.
```
Comment (dengemann) [inline mne/evoked.py]: The eeg channels markeds as bad will be interpolated.

File: mne/evoked.py [hunk 2]
```diff
@@ -938,6 +940,63 @@ def _get_evoked_node(fname):
...
+        raise ValueError("Not all the elements in list are evoked data")
+
+    if interpolate_bads != 'eeg':
+        raise ValueError('Only EEG channels can be interpolated currently.')
+
+    # Copy channels to leave the original evoked datasets intact.
+    all_evokeds = [e.copy() for e in all_evokeds]
+
+    # Interpolates if necessary
+    if interpolate_bads is not None:
```
Comment (dengemann) [inline mne/evoked.py]: == 'eeg'

File: mne/channels/interpolation.py [hunk 3]
```diff
@@ -117,20 +117,29 @@ def _interpolate_bads_eeg(inst):
...
-        raise ValueError('Data must be preloaded')
+        raise ValueError('Data must be preloaded.')
 
     bads_idx = np.array([ch in inst.info['bads'] for ch in inst.ch_names])
-    goods_idx = np.invert(bads_idx)
+    goods_idx = np.zeros(len(inst.ch_names), dtype=np.bool)
 
     picks = pick_types(inst.info, meg=False, eeg=True, exclude=[])
+    goods_idx[picks] = True
+    goods_idx[bads_idx] = False
```
Comment (dengemann) [inline mne/channels/interpolation.py]: does this fix a bug? what was wrong?

**Comment by dengemann** [comment]
@agramfort I would expose the new function in the mne namespace and put the reference entry next to    equalize_channels about line 427

File: mne/channels/interpolation.py (hunk 3 above)
Comment (agramfort) [inline mne/channels/interpolation.py]: yes. Otherwise you can have bads_idx of left n_channels but pos of length n_eeg

**Comment by agramfort** [comment]
comments addressed.

**Comment by dengemann** [comment]
Thanks @agramfort and @MadsJensen !

**Pull request merged**